### ISSUE TITLE
[Bug?]: Extend Solid config causes TS Errors

### ISSUE THREAD

**Issue by ryoid** [solidjs/solid-start#1464]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Extending solid config has required fields that should be optional.

```ts
// app.config.ts
export default defineConfig({
  solid: {
    // Uncomment to supress errors
    // dev: true,
    // ssr: false,
    // hot: true,
    // solid: {
    //   omitNestedClosingTags: false,
    // },
    babel: {
      env: {
        development: {
          // https://github.com/nksaraf/vinxi/issues/204
          compact: true,
        },
      },
    },
  },
});
```

### Expected behavior 🤔

Should not error since these fields have default values. Looking at `vite-plugin-solid` it [seems like these are optional there](https://github.com/solidjs/vite-plugin-solid/blob/873f4cec4db1dcffac9d909191cf828a9902a418/src/index.ts#L41-L55).

### Steps to reproduce 🕹

https://stackblitz.com/edit/github-e58cde-3el8ng?file=package.json

Steps:

1. Extend the Solid config in `app.config.ts` with some babel settings
2. Observe typescript errors

### Context 🔦

Encountered when trying to use config from https://github.com/nksaraf/vinxi/issues/204#issuecomment-2053760779

**Comment by ryansolid** [comment]
Yeah we fixed this in 2.10.x of Vite Plugin Solid but Start hasn't been updated to it. Thanks for pointing this out.

**Comment by ryansolid** [comment]
This has been fixed in https://github.com/solidjs/solid-start/commit/7ae0541097f95fd9e1b7368368794c5cac71ffe8
